File: backend/src/main/java/pap/project/users/characters/model/UserCharacterData.java
package pap.project.users.characters.model;

import org.springframework.lang.NonNull;

public record UserCharacterData(
        @NonNull CharacterType characterType,
//       XXX maybe commented data should be in frontend?
//        int damage,
//        int health,
        int level,
//        int requiredCopiesForNextLevel,
        int currentCopiesCount
) { }

File: frontend/src/app/user-chararacters/character-in-list/user-character-data.ts
import { CharacterType } from '../../user-data/user-data-service';

export interface UserCharacterData {
	readonly characterType: CharacterType;
	readonly unlockedData?: UserCharacterUnlockedData;
}

export interface UserCharacterUnlockedData {
	readonly level: number;
	readonly requiredCopiesForNextLevel: number;
	readonly currentCopiesCount: number;
}
